frontend/src/lib/components/DateFilter/DateFilter.tsx

```diff
@@ -127,19 +136,30 @@ export function DateFilter({
                     return menu
                 }
             }}
+            {...selectProps}
         >
             {[
                 ...Object.entries(dateOptions).map(([key, { inactive }]) => {
                     if (key === 'Custom' && !showCustom) {
                         return null
                     }
 
-                    if (inactive && parsedValue !== key) {
+                    if (inactive && currKey !== key) {
                         return null
                     }
 
                     return (
-                        <Select.Option key={key} value={key} label={makeLabel ? makeLabel(key) : undefined}>
+                        <Select.Option
```

Comment:
I don't know the context for this change, but it feels weird to select an item in a drop down and then not have the drop down show the thing I clicked it. If we want to show the full date string, I wonder if it should be shown along with the range that the user clicked?